src/node/Main.hs

```diff
@@ -87,11 +73,31 @@ baseParams loggingTag args@Args {..} = do
         , bpKademliaDump = kademliaDumpPath
         }
 
+getNodeSystemStart :: (MonadIO m) => Timestamp -> m Timestamp
+getNodeSystemStart cliOrConfigSystemStart
+    | cliOrConfigSystemStart >= 1400000000 =
+    -- UNIX time 1400000000 is Tue, 13 May 2014 16:53:20 GMT.
+    -- It was chosen arbitrarily as some date far enough in the past.
+    pure cliOrConfigSystemStart
+getNodeSystemStart cliOrConfigSystemStart = do
```

Comment:
@gromakovsky 
```
src/node/Main.hs:83:1: Suggestion: Redundant guard
Found:
  getNodeSystemStart cliOrConfigSystemStart
    | otherwise =
      do let frameLength = timestampToSeconds cliOrConfigSystemStart
         currentPOSIXTime <- liftIO $ round <$> getPOSIXTime
         let currentFrame = currentPOSIXTime `div` frameLength
             t = currentFrame * frameLength + (frameLength `div` 2)
         pure $ Timestamp $ sec $ fromIntegral t
    where timestampToSeconds :: Timestamp -> Integer
          timestampToSeconds
            = (`div` 1000000) . toMicroseconds . getTimestamp
Why not:
  getNodeSystemStart cliOrConfigSystemStart
    = do let frameLength = timestampToSeconds cliOrConfigSystemStart
         currentPOSIXTime <- liftIO $ round <$> getPOSIXTime
         let currentFrame = currentPOSIXTime `div` frameLength
             t = currentFrame * frameLength + (frameLength `div` 2)
         pure $ Timestamp $ sec $ fromIntegral t
    where timestampToSeconds :: Timestamp -> Integer
          timestampToSeconds
            = (`div` 1000000) . toMicroseconds . getTimestamp

1 hint (438 ignored)
```